sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -236,13 +310,18 @@
      * Sends a set of events to the associated Event Hub using a batched approach. If the size of events exceed the
      * maximum size of a single batch, an exception will be triggered and the send will fail. By default, the message
      * size is the max amount allowed on the link.
+     *
      * @param events Events to send to the service.
      * @param options The set of options to consider when sending this batch.
      *
      * @return A {@link Mono} that completes when all events are pushed to the service.
      */
     public Mono<Void> send(Iterable<EventData> events, SendOptions options) {
-        Objects.requireNonNull(events, "'options' cannot be null.");
+        if (events == null) {
+            return Mono.error(logger.logExceptionAsError(new NullPointerException("'events' cannot be null.")));
```

Comment:
FluxUtils in azure-core has a convenience method for this `monoError(logger, ex)`